WIP: Extra QBS project generator

stanionascu opened the pull request:
Patches add support for generating basic [qbs](http://doc.qt.io/qbs/) project files to be used with QtCreator.
Ninja/NMake/Make has to be used though for building the project and the IDE must be configured to use the corresponding build tools.

P.S.
Know that using the mailing list is preferred, but find it a bit cumbersome.
Would be nice to get some input before submitting the patches to the mailing list.

bradking commented:
Thanks for working on this.  Please squash this down to a single commit.  We don't need to review your development history ;)

stanionascu commented:
Squashed. Thanks for checking.

bradking commented:
Thanks.

* Please add a file to `Help/generators/` and update `Help/manual/cmake-generators.7.rst` to document the new generator.

* Please add a  `Help/release/dev/add-extra-qbs-generator.rst` file with a release note.  See adjacent files for examples.

* Please look at `Tests/CMakeLists.txt` for the handling of other extra generators like `CodeBlocks` and add a similar smoke test for this one too.

* Please convert the new source files to LF newlines instead of CRLF.  Also please wrap C++ source lines to <= 79 columns.

* Please use the `{`-aligned-with-contents indentation style present in the other CMake sources.

stanionascu commented:
Thanks again for checking. :)

* Rebased on top of current master.
* Updated code to API changes.
* Added `Help/generator/Qbs.rst`
* Updated `Help/manual/cmake-generators.7.rst`
* Added `Help/release/dev/add-extra-qbs-generator.rst`
* Updated `Tests/CMakeLists.txt` with `Simple_QbsGenerator` test.
* Converted from CRLF to LF (where applicable).
* IDE was and still is set at 80 columns, so re-wrapped where applicable.
* Aligned `{` with contents.

bradking commented:
Great, that's in pretty good shape.  I just have a couple nitpicks:

* Please make the section header underlines match the title lengths in `.rst` files.
* In the release note, use

        :generator:`Qbs`
   to to link to the generator directly.

stanionascu commented:
Thanks.

As requested:
* Rebased on top of current master.
* Updated section header underlines.
* Linked the generator in release note.

bradking commented:
Thanks.  I've applied this as commit f85db2f32358e6de921aba7d1cb8ecb81da934c0 with a minor tweak to the commit message.  Further revisions may be squashed after we see how it goes on nightly testing.

bradking commented:
I'm sorry to do this after originally accepting this change, but I've decided to revert this generator pending further discussion on ways to make CMake and QtCreator work better together.  See commit deec97d8eca4db67be09031757fd11f66c1a037b.  As noted in `CONTRIBUTING.rst` the [mailing list](http://www.cmake.org/mailman/listinfo/cmake-developers) is the preferred contribution path.  It will allow broader discussion.

stanionascu commented:
No wonder, as it was a wip ;)

As anyway I planned to do more work on it for specific cases, so it's good that you've reverted it (the complete idea was to replace later in QtCreator the CMakeCbpParser with the QbsParser in an attempt to make CMake in QtCreator support better).

Will use the mailing-list next time.

Thanks!
